### PR TITLE
Added UnitTests to the project

### PR DESCRIPTION
**Pull request by StraussLukas**

### CODE REVIEW

File: src/test/java/stage1/RunnerSprintNameGeneratorTest.java [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+		RunnerSprintNameGenerator.getSprintNames().get(1).voteUp();
+		RunnerSprintNameGenerator.getSprintNames().get(2).voteUp();
+		RunnerSprintNameGenerator.getSprintNames().get(2).voteUp();
+		RunnerSprintNameGenerator.markNamesDraw(RunnerSprintNameGenerator.getNamesWithMostVotes());
+		assertEquals(Color.YELLOW, RunnerSprintNameGenerator.getNamesWithMostVotes().get(0).getOutputName().getBackground());
+		assertEquals(Color.YELLOW, RunnerSprintNameGenerator.getNamesWithMostVotes().get(1).getOutputName().getBackground());
+	
+	}
+	@Test
+	public void test_markWinnerName_checkIfCorrectNameIsHighlightedGreen() {
```
Comment (StraussLukas) [inline src/test/java/stage1/RunnerSprintNameGeneratorTest.java:213]: UnitTest not successful

File: src/test/java/stage1/RunnerSprintNameGeneratorTest.java [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+		RunnerSprintNameGenerator.getSprintNames().get(2).voteUp();
+		RunnerSprintNameGenerator.getSprintNames().get(2).voteUp();
+		RunnerSprintNameGenerator.getSprintNames().get(2).voteUp();
+		RunnerSprintNameGenerator.startRevote(RunnerSprintNameGenerator.getNamesWithMostVotes());
+		assertEquals(1, RunnerSprintNameGenerator.getSprintNames().get(0).getIndex());
+		assertEquals(2, RunnerSprintNameGenerator.getSprintNames().get(1).getIndex());
+	}
+	
+	@Test
+	public void test_markNamesDraw_checkIfNamesWereHighlightedYellow() {
```
Comment (StraussLukas) [inline src/test/java/stage1/RunnerSprintNameGeneratorTest.java:200]: UnitTest not successful

File: src/test/java/stage1/RunnerSprintNameGeneratorTest.java (hunk 1 above)
Comment (GeroHerkenrath) [reply src/test/java/stage1/RunnerSprintNameGeneratorTest.java:213]: Well, that's to be expected from the logic you wrote. :)
So in line 253 in this test you call `RunnerSprintNameGenerator.getNamesWithMostVotes()`. If you look up that method, it executes `clearHighlighting()` on each `SprintName` object the runner has in its static member array, so the background color that was set in `markWinnerName(RunnerSprintNameGenerator.getNamesWithMostVotes())` gets reset to white again.

In the broader context that's the result of mixing something like a getter (more or less) and UI update code.
For now you can resolve this by not getting the sprint name you are interested in via `getNamesWithMostVotes().get(0)` (in the assert), but instead via `getSprintNames().get(1)`. That does not change the color and will make the test pass

File: src/test/java/stage1/RunnerSprintNameGeneratorTest.java (hunk 2 above)
Comment (GeroHerkenrath) [reply src/test/java/stage1/RunnerSprintNameGeneratorTest.java:200]: I haven't checked due to time yet, but I am pretty sure that this is due to the same issue as in test `test_markWinnerName_checkIfCorrectNameIsHighlightedGreen`, so it should be resolvable via the same means.

**Review by GeroHerkenrath — CHANGES_REQUESTED** [review]
Once both tests are repaired finish the draft and I will review again (maybe also in peson with you).

**Review by GeroHerkenrath — APPROVED** [review]

**Pull request merged**